### PR TITLE
handle case where form is already saved

### PR DESCRIPTION
**Pull request by snopoke**
##### SUMMARY
When errors happen after the form is saved the error handling raises another error when trying to save the already saved form:

https://sentry.io/organizations/dimagi/issues/1249436013

### CODE REVIEW

**Review by millerdev — APPROVED** [review]
Are there tests the situation this is testing? If not, would be great if they were added.

**Comment by snopoke** [comment]
@millerdev there are no tests right now. I'll add some

File: corehq/apps/receiverwrapper/tests/test_submit_errors.py [hunk 1]
```diff
@@ -276,6 +277,28 @@ def test_error_publishing_to_kafka(self):
         self.assertFalse(form.is_error)
         self.assertTrue(form.initial_processing_complete)
 
+    def test_error_writing_to_blobdb(self):
+        error = ConnectionClosedError(endpoint_url='url')
+        with self.assertRaises(ConnectionClosedError), patch.object(get_blob_db(), 'put', side_effect=error):
+            self._submit('form_with_case.xml')
+
+        stubs = UnfinishedSubmissionStub.objects.filter(domain=self.domain, saved=False).all()
+        self.assertEqual(1, len(stubs))
```
Comment (millerdev) [inline corehq/apps/receiverwrapper/tests/test_submit_errors.py]: nit: This is vulnerable to failure if another test does not clean up after itself, leaving other stubs in the db. Does this stub have an id (possibly _ad38211be256653bceac8e2156475666_)? If it does, then one way to work around the vulnerability would be to check if a stub exists having that expected id.

File: corehq/form_processor/models.py [hunk 2]
```diff
@@ -547,13 +548,13 @@ def get_data(self, path):
     def get_xml(self):
         try:
             return self.get_attachment('form.xml')
-        except NotFound:
+        except AttachmentNotFound:
```
Comment (millerdev) [inline corehq/form_processor/models.py]: Seems like `NotFound` could still be raised here (ex. if the blob is missing)? Does it makes sense to catch both?
```suggestion
        except (AttachmentNotFound, NotFound):
```

File: corehq/form_processor/models.py [hunk 3]
```diff
@@ -547,13 +546,13 @@ def get_data(self, path):
...
             raise MissingFormXml(self.form_id)
 
     def xml_md5(self):
         try:
             return self.get_attachment_meta('form.xml').content_md5()
-        except NotFound:
+        except AttachmentNotFound:
```
Comment (millerdev) [inline corehq/form_processor/models.py]: Same as above

Comment (snopoke) [reply corehq/form_processor/models.py]: updated, thanks

File: corehq/apps/receiverwrapper/tests/test_submit_errors.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/receiverwrapper/tests/test_submit_errors.py]: done

Comment (snopoke) [reply corehq/apps/receiverwrapper/tests/test_submit_errors.py]: updated

**Comment by snopoke** [comment]
@millerdev ready for review again

**Review by millerdev — APPROVED** [review]

**Pull request merged**